src/libraries/System.Reflection.Emit/tests/PersistableAssemblyBuilder/AssemblySaveTestsWithVariousTypes.cs

```diff
@@ -0,0 +1,455 @@
+                    Assert.Equal(sourceType.Attributes, typeFromDisk.Attributes);
+
+                    MethodInfo[] sourceMethods = sourceType.GetMethods();
+                    MethodInfo[] methodsFromDisk = typeFromDisk.GetMethods();
+
+                    Assert.True(methodsFromDisk.Length <= 1 );
+
+                    // Method comparisons
+                    if (sourceMethods.Length > 0)
+                    {
+                        MethodInfo sourceMethod = sourceMethods[0];
+                        MethodInfo methodFromDisk = methodsFromDisk[0];
+
+                        Assert.Equal(sourceMethod.Name, methodFromDisk.Name);
+                        Assert.Equal(sourceMethod.Attributes, methodFromDisk.Attributes);
+                        Assert.Equal(sourceMethod.ReturnType.FullName, methodFromDisk.ReturnType.FullName);
+                    }
+                }
+            }
+        }
+
+        [Fact]
+        public void TwoInterfaceManyMethodsThenNone()
+        {
+            using (TempFile file = TempFile.Create())
+            {
+                Type[] types = new Type[] { typeof(IMultipleMethod), typeof(INoMethod2) };
+
+                AssemblyTools.WriteAssemblyToDisk(s_assemblyName, types, file.Path);
+
+                Assembly assemblyFromDisk = AssemblyTools.TryLoadAssembly(file.Path);
+
+                Assert.NotNull(assemblyFromDisk);
+
+                Module moduleFromDisk = assemblyFromDisk.Modules.First();
+                Type[] typesFromDisk = moduleFromDisk.GetTypes();
+
+                Assert.NotNull(moduleFromDisk);
+                Assert.Equal(types.Length, typesFromDisk.Length);
+
+                // Type comparisons
+                for (int i = 0; i < types.Length; i++)
+                {
+                    Type sourceType = types[i];
+                    Type typeFromDisk = typesFromDisk[i];
+
+                    Assert.Equal(sourceType.Name, typeFromDisk.Name);
+                    Assert.Equal(sourceType.Namespace, typeFromDisk.Namespace);
+                    Assert.Equal(sourceType.Attributes, typeFromDisk.Attributes);
+
+                    MethodInfo[] sourceMethods = sourceType.GetMethods();
+                    MethodInfo[] methodsFromDisk = typeFromDisk.GetMethods();
+
+                    Assert.Equal(sourceMethods.Length, methodsFromDisk.Length);
+
+                    // Method comparisons
+                    for (int j = 0; j < sourceMethods.Length; j++)
+                    {
+                        MethodInfo sourceMethod = sourceMethods[j];
+                        MethodInfo methodFromDisk = methodsFromDisk[j];
+
+                        Assert.Equal(sourceMethod.Name, methodFromDisk.Name);
+                        Assert.Equal(sourceMethod.Attributes, methodFromDisk.Attributes);
+                        Assert.Equal(sourceMethod.ReturnType.FullName, methodFromDisk.ReturnType.FullName);
+                    }
```

Comment:
The structure of each test is almost exactly the same. The only difference between different tests is the list of types that the test should emit. Can we reduce the duplication?